app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/components/MenuItem.kt

```diff
@@ -50,18 +52,33 @@ fun MenuItem(
             textAlign = TextAlign.Start,
             modifier = Modifier.weight(1f)
         )
-        if(icon != null) {
-            Icon(
-                imageVector = icon,
-                contentDescription = iconContentDescription,
-                tint = if(enabled)
+        if(vectorIcon != null || painterIcon != null) {
+            vectorIcon?.let {
+                Icon(
+                    imageVector = vectorIcon,
+                    contentDescription = iconContentDescription,
+                    tint = if (enabled)
                         MaterialTheme.colors.surface
                     else
                         MaterialTheme.colors.surface.copy(alpha = 0.2f),
-                modifier = Modifier
-                    .padding(DP.tiny)
-                    .align(Alignment.CenterVertically)
-            )
+                    modifier = Modifier
+                        .padding(DP.tiny)
+                        .align(Alignment.CenterVertically)
+                )
+            }
+            painterIcon?.let {
+                Icon(
+                    painter = painterIcon,
+                    contentDescription = iconContentDescription,
+                    tint = if (enabled)
+                        MaterialTheme.colors.surface
+                    else
+                        MaterialTheme.colors.surface.copy(alpha = 0.2f),
+                    modifier = Modifier
+                        .padding(DP.tiny)
+                        .align(Alignment.CenterVertically)
+                )
+            }
         }
```

Comment:
The outer if-check seems unnecessary to me

Reply:
Not sure what you mean here. You mean the check for `painterIcon` or `vectorIcon`? The `Icon` composable only allows for one or the other. You cant have both. The compiler gives an error.